### ISSUE TITLE
How can I load a gem with RubyGateway?

### ISSUE THREAD

**Issue by mshibanami** [johnfairh/RubyGateway#35]
Hi @johnfairh,

Thanks for maintaining this great framework.
I'm now trying to use it but stuck at the very beginning.

Specifically, at this code:

> try! Ruby.require(filename: "rouge")

I ran the above code, then got this error:

> Thread 1: Fatal error: 'try!' expression unexpectedly raised an error: Ruby exception: LoadError: cannot load such file -- rouge

I did `gem install rouge` or installed it via Bundler but nothing changed.
Would you please give me your thoughts for solving this?

---

By the way, this repo has a unit test for calling rouge in  `Tests/RubyGatewayTests/TestDemo.swift`. The test result is green but I noticed that it doesn't fail when rouge can't be loaded.

So I forked this repo and added this small change:

https://github.com/mshibanami/RubyGateway/commit/ca54f3aceea808906e2523c591676a6d9d26954a

I then ran the tests via GitHub Actions as this repo does, then it failed on all environments:

https://github.com/mshibanami/RubyGateway/runs/3385872893?check_suite_focus=true#step:4:199

So I wonder if this repo neither tests it correctly. What do you think?

**Comment by johnfairh** [comment]
In my experience, this problem happens because RubyGateway is not using the same version of Ruby as the one that the gem is installed to.  You don't say what OS you are using, but on macOS the module defaults to using the system Ruby that comes with the base Apple OS.  See [the readme](https://github.com/johnfairh/RubyGateway#installation) for how to change this.

Thanks for the pointers to this happening in CI.

The Linux failures were because of image changes and my very fragile script -- I fixed those and the rouge test passed fine.

The macOS failure was because I misunderstood how the image was configured: I thought its `ruby` was the macOS system Ruby.  But in fact it is something else installed into `/usr/local`.  So the `gem install rouge` was installing the gem into that Ruby, but then the `swift test` was invoking RubyGateway pointing at the macOS system Ruby, which did not have the gem.

I think I made `testRouge()` hide failures because I couldn't express a dependency from the swift package to the gem.  This issue shows it is probably better to fail, so I've made that change too, #36.
